### ISSUE TITLE
Remove unnecessary null pointer checks

### ISSUE THREAD

**Issue by elfring** [bagder/curl#149]
[An extra null pointer check is not needed](http://pubs.opengroup.org/onlinepubs/9699919799/functions/free.html "Do I need to check for NULL before free(p)?") in functions like the following.
* [freecookie](https://github.com/bagder/curl/blob/df5578a7a304a23f9aa3670daff8573ec3ef416f/lib/cookie.c#L104)
* [Curl_wildcard_dtor](https://github.com/bagder/curl/blob/df5578a7a304a23f9aa3670daff8573ec3ef416f/lib/wildcard.c#L45)
* [unit_stop](https://github.com/bagder/curl/blob/6f8046f7a4bd3d6edcc53c2eec936105ec424d54/tests/unit/unit1305.c#L62)

Would you like to apply the following [semantic patch](http://coccinelle.lip6.fr/ "Program transformation with Coccinelle?") to find more update candidates?
```
@Remove_unnecessary_pointer_checks@
expression x;
@@
-if (x)
    free(x);
```

**Comment by bagder** [comment]
I agree they're unnecessary but also harmless. If you or someone brings a patch to remove them, I'll appreciate it. I'm not gonna spend many cycles of my own to remove these.

**Comment by bagder** [comment]
Fixed merged in commit 9e661601feba
